server/main.lua

```diff
@@ -154,75 +126,219 @@ function GetEmployees(src, accountName, type)
+		if not success then
+			exports.qbx_core:Notify(src, 'Unable to fire citizen.', 'error')
+			return
+		end
+	
+		local notifyMessage = data.type == 'gang' and 'You have been expelled from the gang!' or 'You have been fired! Good luck.'
+		TriggerEvent('qb-log:server:CreateLog', logArea..'menu', data.type..' Fire', 'orange', playerFullName .. ' successfully fired ' .. employeeFullName .. ' (' .. organizationLabel .. ')', false)
+		exports.qbx_core:Notify(src, logType, 'success')
+		exports.qbx_core:Notify(employee.PlayerData.source, notifyMessage, 'error')
+	else
+		local offlineEmployee = MySQL.query.await('SELECT * FROM players WHERE citizenid = ? LIMIT 1', {data.source})
+		if not offlineEmployee[1] then
+			exports.qbx_core:Notify(src, 'Civilian doesn\'t exist?', 'error')
+			return
+		end
+	
+		employee = offlineEmployee[1]
+		employee[data.type] = json.decode(employee[data.type])
+		employee.charinfo = json.decode(employee.charinfo)
+		local employeeFullName = employee.charinfo.firstname .. ' ' .. employee.charinfo.lastname
+	
+		if employee[data.type].grade.level > player.PlayerData.gang.grade.level then
+			exports.qbx_core:Notify(src, 'You cannot fire this citizen!', 'error')
+			return
+		end
+	
+		local role = {
+			name = data.type == 'gang' and 'none' or 'unemployed',
+			label = data.type == 'gang' and gangs['none'].label or jobs['unemployed'].label,
+			payment = data.type == 'gang' and 0 or jobs['unemployed'].grades[0].payment,
+			onduty = data.type ~= 'gang',
+			isboss = false,
+			grade = {
+				name = data.type == 'gang' and gangs['none'].grades[0].name or jobs['unemployed'].grades[0].name,
+				level = 0
+			}
+		}
+	
+		local updateColumn = data.type == 'gang' and 'gang' or 'job'
+		MySQL.update(string.format('UPDATE players SET %s = ? WHERE citizenid = ?', updateColumn), {json.encode(role), data.source})
+		exports.qbx_core:Notify(src, logType, 'success')
+		TriggerEvent('qb-log:server:CreateLog', logArea..'menu', data.type..' Fire', 'orange', playerFullName .. ' successfully fired ' .. employeeFullName .. ' (' .. organizationLabel .. ')', false)
+	end
+	TriggerClientEvent('qbx_management:client:OpenMenu', src, data.type)
+end)
+
+---@param type 'job'|'gang'
+---@param amount number
+RegisterNetEvent('qbx_management:server:withdrawMoney', function(type, amount)
+	local src = source
+	local success, player, account = WithdrawMoney(src, amount, type, type == 'gang' and 'Gang menu withdraw' or 'Boss menu withdraw')
+	if not success or not player then return end
+	local logArea = type == 'gang' and 'gang' or 'boss'
+	
+	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Withdraw Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully withdrew $' .. amount .. ' (' .. account .. ')', false)
+	TriggerClientEvent('qbx_management:client:OpenMenu', src, type)
+end)
```

Comment:
```suggestion
---@param group 'job'|'gang'
---@param amount number
RegisterNetEvent('qbx_management:server:withdrawMoney', function(group, amount)
	local success, player, account = WithdrawMoney(source, amount, group, group == 'gang' and 'Gang menu withdraw' or 'Boss menu withdraw')
	if not success or not player then return end
	local logArea = group == 'gang' and 'gang' or 'boss'
	
	TriggerEvent('qb-log:server:CreateLog', logArea..'menu', 'Withdraw Money', 'yellow', player.PlayerData.charinfo.firstname .. ' ' .. player.PlayerData.charinfo.lastname .. ' successfully withdrew $' .. amount .. ' (' .. account .. ')', false)
	TriggerClientEvent('qbx_management:client:OpenMenu', source, type)
end)
```

here we want to rename type to group, to keep going with the naming scheme above.

I also removed the local src = source here because we do not have a yield in any of these functions.

Reply:
done